code/modules/vueui/ui.dm

```diff
@@ -90,7 +90,13 @@ main ui datum.
 
 	var/params = "window=[windowid];file=[windowid];titlebar=0;can_resize=0;"
 	if(width && height)
-		params += "size=[width]x[height];"
+		var/winsize = winget(user, "mapwindow", "size")
+		var/map_size = splittext(winsize, "x")
+		var/screen_height = text2num(map_size[1])
+		if(height > screen_height)
+			params += "size=[width]x[screen_height - 100];"
```

Comment:
What happens if we're for some reason playing with a screen that has <100 height? Does a negative height do something weird?